grpc/src/main/java/io/stargate/grpc/metrics/api/GrpcMetricsTagProvider.java

```diff
@@ -29,7 +30,8 @@ public interface GrpcMetricsTagProvider {
    * Prometheus does not allow different tags from the single process.
    *
    * @param call Server-side gRPC call
+   * @param requestHeaders Call metadata
    * @return Tags
    */
-  Tags getCallTags(ServerCall<?, ?> call);
+  Tags getCallTags(ServerCall<?, ?> call, Metadata requestHeaders);
```

Comment:
I guess/hope there are no external developers implementing this interface; if there were, this would be backwards-incompatible change (could probably make it work by deprecating old method, having new method have default implementation that called old method).

This may also mean that this should only be done in 2.x? (there may be Stargate 1.x developers that do this)

Reply:
Wait what are we talking about here.. This is interface? It's fully backward compatible, when someone updates the SG version he will need to change the signature of the existing interfaces.. That has nothing to do with backward compatibility.. Every time we update Quarkus there is some change that needs to be applied, that's normal.. 

It would not be backward compatible if this change would cause the existing SG installations not to be usable anymore, or break in some point.. This is not the case here.. Let's not mix things up..

Reply:
In practice not a big deal, but conceptually this would be breaking change -- for minor version, Quarkus upgrade actually should never require a code change. That's what semantic version says. It happens given Quarkus is still sort of young project, but compilation breaking changes really aren't supposed to go in minor (or esp. patch versions).

The way to handle this -- if it mattered, and I agree that it doesn't :) -- would be to leave old version as deprecated, redirect to new (or vice versa).